go/tasks/plugins/array/k8s/transformer.go

```diff
@@ -99,6 +139,39 @@ func FlyteArrayJobToK8sPodTemplate(ctx context.Context, tCtx core.TaskExecutionC
 			Annotations:     annotations,
 			OwnerReferences: []metav1.OwnerReference{tCtx.TaskExecutionMetadata().GetOwnerReference()},
 		},
-		Spec: *podSpec,
-	}, arrayJob, nil
+	}
+	if taskTemplate.GetContainer() != nil {
+		podSpec, err := flytek8s.ToK8sPodSpec(ctx, arrTCtx)
+		if err != nil {
+			return v1.Pod{}, nil, err
+		}
+		pod.Spec = *podSpec
+	} else if taskTemplate.GetK8SPod() != nil {
+		k8sPod, err := buildPodMapTask(taskTemplate, tCtx.TaskExecutionMetadata())
+		if err != nil {
+			return v1.Pod{}, nil, err
+		}
+		pod.Labels = utils.UnionMaps(pod.Labels, k8sPod.Labels)
+		pod.Annotations = utils.UnionMaps(pod.Annotations, k8sPod.Annotations)
+		pod.Spec = k8sPod.Spec
+
+		// Here we templatize the k8sPod primary container args. The call to ToK8sPodSpec for the task container target
+		// case already handles this but we must explicitly do so for K8sPod task targets.
+		containerIndex, err := getTaskContainerIndex(&pod)
+		if err != nil {
+			return v1.Pod{}, nil, err
+		}
+		pod.Spec.Containers[containerIndex].Args, err = template.Render(ctx, pod.Spec.Containers[containerIndex].Args,
```

Comment:
I think we should do this on `Command` as well... like we do in `ToK8sPod`

Reply:
done